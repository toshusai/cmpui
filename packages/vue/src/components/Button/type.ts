export const ButtonVariant = {
  primary: "primary",
  secondary: "secondary",
} as const;

export type ButtonVariant = (typeof ButtonVariant)[keyof typeof ButtonVariant];

export const ButtonSize = {
  S: "S",
  M: "M",
  L: "L",
} as const;

export type ButtonSize = (typeof ButtonSize)[keyof typeof ButtonSize];

export type ButtonProps = {
  variant: ButtonVariant;
  size: ButtonSize;
  disabled: boolean;
};
