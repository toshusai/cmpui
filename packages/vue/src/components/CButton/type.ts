export const CButtonVariant = {
  primary: "primary",
  secondary: "secondary",
} as const;

export type CButtonVariant =
  (typeof CButtonVariant)[keyof typeof CButtonVariant];

export const CButtonSize = {
  S: "S",
  M: "M",
  L: "L",
} as const;

export type CButtonSize = (typeof CButtonSize)[keyof typeof CButtonSize];

export type CButtonProps = {
  variant?: CButtonVariant;
  size?: CButtonSize;
  disabled?: boolean;
};
