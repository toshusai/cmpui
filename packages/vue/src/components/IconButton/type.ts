export const IconButtonSize = {
  S: "S",
  M: "M",
} as const;

export type IconButtonSize =
  (typeof IconButtonSize)[keyof typeof IconButtonSize];

export type IconButtonProps = {
  bordered?: boolean;
  disabled?: boolean;
  rounded?: boolean;
  selected?: boolean;
  size?: IconButtonSize;
};
