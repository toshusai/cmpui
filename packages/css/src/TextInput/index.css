.cmpui_text-input__root {
  background-color: var(--cmpui-bg-color);
  border: 1px solid var(--cmpui-border-color);

  align-items: center;
  border-radius: 4px;
  box-sizing: border-box;
  display: flex;
  height: 32px;
  justify-content: flex-start;
  position: relative;
  text-decoration: none;
}

.cmpui_text-input__root:focus-within {
  outline: var(--cmpui-primary-color) solid 1px;
  outline-offset: 1px;
}

.cmpui_text-input__root[aria-disabled="true"] {
  cursor: default;
  opacity: 0.5;
}

.cmpui_text-input__root[aria-invalid="true"] {
  border: 1px solid var(--cmpui-error-color);
}
.cmpui_text-input__root[aria-invalid="true"]:hover {
  border: 1px solid var(--cmpui-error-color);
}

.cmpui_text-input__root:hover {
  border: 1px solid var(--cmpui-border-hover-color);
}

.cmpui_text-input__root[aria-disabled="true"]:hover {
  border: 1px solid var(--cmpui-border-color);
}

.cmpui_text-input__label {
  background-image: linear-gradient(
    to bottom,
    transparent 10%,
    var(--cmpui-bg-color) 40%,
    var(--cmpui-bg-color) 60%,
    transparent 60%
  );
  color: var(--cmpui-text2-color);

  font-size: 12px;
  line-height: 14px;
  padding: 0 4px;
  position: absolute;
  transform: translate(4px, -16px);
}
.cmpui_text-input__root[aria-invalid="true"] > .cmpui_text-input__label {
  color: var(--cmpui-error-color);
}

.cmpui_text-input__input:nth-child(3) {
  padding-left: 0px;
}

.cmpui_text-input__input {
  color: var(--cmpui-text-color);
  font-family: var(--cmpui-font-family);

  background: transparent;
  border: none;
  box-sizing: border-box;
  display: flex;
  font-size: 16px;
  height: 100%;
  line-height: 16px;
  outline: none;
  padding: 0 8px;
  width: 100%;
}

.cmpui_text-input__input::placeholder {
  color: var(--cmpui-text3-color);
  font-style: italic;
}
