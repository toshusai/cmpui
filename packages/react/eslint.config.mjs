"use strict";
import eslintConfigPrettier from "eslint-config-prettier";
import tseslint from "typescript-eslint";
import eslintPluginImport from "eslint-plugin-import";

export default [
  ...tseslint.configs.recommended,
  {
    ignores: ["dist/*", "node_modules/*", "storybook-static/*"],
  },
  {
    plugins: { import: eslintPluginImport },
    rules: {
      "vue/multi-word-component-names": "off",
      "no-unused-vars": "off",
      "@typescript-eslint/no-unused-vars": ["error"],
      "import/order": [
        "error",
        {
          "newlines-between": "always",
        },
      ],
    },
    languageOptions: {
      parserOptions: { parser: tseslint.parser, sourceType: "module" },
    },
  },
  eslintConfigPrettier,
];
