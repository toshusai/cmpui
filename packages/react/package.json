{
  "name": "@toshusai/cmpui",
  "version": "1.0.0-alpha.13",
  "description": "CmpUI is a component library designed for creating complex GUI applications in web browsers.",
  "main": "dist/index.mjs",
  "types": "dist/index.d.mts",
  "scripts": {
    "typecheck": "tsc -p tsconfig.ci.json",
    "test:watch": "vitest",
    "test": "vitest run",
    "dev": "tsup --watch",
    "build": "tsup",
    "storybook": "storybook dev -p 6007",
    "storybook:build": "storybook build",
    "storybook:test": "test-storybook",
    "storybook:test:ci": "concurrently -k -s first -n \"SB,TEST\" -c \"magenta,blue\" \"pnpm storybook:build && npx http-server storybook-static --port 6006 --silent\" \"wait-on http://127.0.0.1:6006 && pnpm storybook:test\"",
    "format": "prettier .",
    "format:fix": "prettier --write .",
    "lint": "eslint .",
    "lint:fix": "eslint --fix .",
    "fix": "npm run format:fix && npm run lint:fix"
  },
  "dependencies": {
    "@floating-ui/dom": "^1.6.10",
    "@radix-ui/react-context-menu": "^2.1.5",
    "@radix-ui/react-dropdown-menu": "^2.0.6",
    "@radix-ui/react-menubar": "^1.0.4",
    "@radix-ui/react-select": "^2.0.0",
    "@radix-ui/react-toast": "^1.1.5",
    "@toshusai/cmpui-core": "link:../core",
    "@toshusai/cmpui-css": "link:../css",
    "react-transition-group": "^4.4.5"
  },
  "peerDependencies": {
    "react": "^18.2.0",
    "react-dom": "^18.2.0"
  },
  "devDependencies": {
    "@playwright/test": "^1.41.2",
    "@storybook/addon-essentials": "8.2.6",
    "@storybook/addon-interactions": "8.2.6",
    "@storybook/addon-links": "8.2.6",
    "@storybook/addon-mdx-gfm": "8.2.6",
    "@storybook/addon-onboarding": "8.2.6",
    "@storybook/addon-themes": "8.2.6",
    "@storybook/blocks": "8.2.6",
    "@storybook/react": "8.2.6",
    "@storybook/react-vite": "8.2.6",
    "@storybook/test": "8.2.6",
    "@storybook/test-runner": "^0.17.0",
    "@storybook/theming": "8.2.6",
    "@tabler/icons-react": "^3.15.0",
    "@types/jest": "^29.5.11",
    "@types/react": "^18.2.37",
    "@types/react-dom": "^18.2.15",
    "@types/react-transition-group": "^4.4.11",
    "concurrently": "^8.2.2",
    "eslint": "^9.9.0",
    "eslint-config-prettier": "^9.1.0",
    "eslint-plugin-import": "^2.29.1",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "storybook": "8.2.6",
    "tsup": "^8.0.1",
    "typescript": "^5.2.2",
    "vitest": "^1.1.0",
    "wait-on": "^7.2.0"
  },
  "keywords": [
    "react",
    "typescript",
    "components",
    "ui",
    "gui"
  ],
  "author": "toshusai",
  "license": "MIT"
}
