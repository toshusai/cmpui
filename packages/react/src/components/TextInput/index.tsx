import { forwardRef, memo, useRef } from "react";

import { mergeRefs } from "../..";
import { classNames } from "../../utils/classNames";
import { useNativeOnChange } from "../../utils/interactions/useNativeOnChange";

import "./index.css";

export type TextInputProps = {
  htmlPrefix?: string;
  invalid?: boolean;
  label?: React.ReactNode;
  placeholder?: string;
  prefix?: React.ReactNode;
  suffix?: React.ReactNode;
  value?: string;
  onNativeChange?: (e: Event) => void;
} & Omit<React.InputHTMLAttributes<HTMLInputElement>, "prefix">;

export const TextInput = memo(
  forwardRef<HTMLInputElement, TextInputProps>(function TextInput(
    {
      className,
      htmlPrefix,
      invalid,
      label,
      placeholder,
      prefix,
      suffix,
      value,
      onNativeChange,
      ...props
    },
    fRef,
  ) {
    const ref = useRef<HTMLInputElement>(null);
    useNativeOnChange(ref, onNativeChange);
    return (
      <div
        className={classNames("cmpui_text-input__root", className)}
        aria-disabled={props.disabled}
        aria-invalid={invalid}
      >
        {prefix}
        {typeof label === "string" ? (
          <label className="cmpui_text-input__label" htmlFor={props.id}>
            {label}
          </label>
        ) : (
          label
        )}
        <input
          ref={mergeRefs(ref, fRef)}
          type="text"
          className="cmpui_text-input__input"
          value={value ?? ""}
          placeholder={placeholder ?? ""}
          prefix={htmlPrefix}
          aria-invalid={invalid}
          {...props}
        />
        {suffix}
      </div>
    );
  }),
);
