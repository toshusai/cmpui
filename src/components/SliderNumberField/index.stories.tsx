import type { Meta, StoryObj } from "@storybook/react";
import { useState } from "react";
import { SliderNumberField } from ".";
import { TextInput } from "../TextInput";

const meta: Meta<typeof SliderNumberField> = {
  title: "Form/SliderNumberField",
  component: SliderNumberField,
};

export default meta;

type Story = StoryObj<typeof SliderNumberField>;

export const Basic: Story = {
  render: function Render(props) {
    const [value, setValue] = useState([0]);
    return (
      <SliderNumberField
        {...props}
        label="value"
        inputMode="decimal"
        value={value}
        onChangeValue={setValue}
      />
    );
  },
};

export const Mixed: Story = {
  render: function Render(props) {
    const [values, setValues] = useState([1, 2, 3]);
    return (
      <div>
        <SliderNumberField
          {...props}
          label="value"
          value={values}
          onChangeValue={(values) => {
            if (!Array.isArray(values)) throw new Error("values is not array");
            setValues(values);
          }}
        />
        <div
          style={{
            display: "flex",
            flexDirection: "column",
            gap: 4,
            marginTop: 8,
          }}
        >
          {values.map((value, index) => (
            <TextInput
              type="number"
              key={index}
              value={value.toString()}
              onChange={(e) => {
                const newValues = [...values];
                newValues[index] = Number(e.target.value);
                setValues(newValues);
              }}
            />
          ))}
        </div>
      </div>
    );
  },
};
