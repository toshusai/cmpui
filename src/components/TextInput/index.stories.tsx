import type { Meta, StoryObj } from "@storybook/react";
import { useState } from "react";
import { TextInput } from ".";

const meta: Meta<typeof TextInput> = {
  title: "Components/Form/TextInput",
  component: TextInput,
  args: {
    label: "Label",
  },
};

export default meta;
type Story = StoryObj<typeof TextInput>;

export const Basic: Story = {
  render: function Render() {
    const [value, setValue] = useState("");
    return (
      <TextInput
        label="Label"
        placeholder="Placeholder"
        value={value}
        onChange={(event) => setValue(event.target.value)}
      />
    );
  },
};

export const Label: Story = {
  render: function Render() {
    const [value, setValue] = useState("");
    return (
      <TextInput
        label="Label"
        id="text-input-label"
        value={value}
        onChange={(event) => setValue(event.target.value)}
      />
    );
  },
};

export const Placeholder: Story = {
  render: function Render() {
    const [value, setValue] = useState("");
    return (
      <TextInput
        label=""
        placeholder="Placeholder"
        value={value}
        onChange={(event) => setValue(event.target.value)}
      />
    );
  },
};

export const Suffix: Story = {
  render: function Render() {
    const [value, setValue] = useState("");
    return (
      <TextInput
        label="Label"
        prefix={
          <svg
            xmlns="http://www.w3.org/2000/svg"
            height="24"
            viewBox="0 -960 960 960"
            width="24"
            style={{
              maxWidth: 16,
              height: 16,
            }}
          >
            <title>search</title>
            <path d="M784-120 532-372q-30 24-69 38t-83 14q-109 0-184.5-75.5T120-580q0-109 75.5-184.5T380-840q109 0 184.5 75.5T640-580q0 44-14 83t-38 69l252 252-56 56ZM380-400q75 0 127.5-52.5T560-580q0-75-52.5-127.5T380-760q-75 0-127.5 52.5T200-580q0 75 52.5 127.5T380-400Z" />
          </svg>
        }
        value={value}
        onChange={(event) => setValue(event.target.value)}
      />
    );
  },
};

export const Disabled: Story = {
  render: function Render() {
    const [value, setValue] = useState("");
    return (
      <TextInput
        label="Label"
        value={value}
        onChange={(event) => setValue(event.target.value)}
        disabled
      />
    );
  },
};

export const Invalid: Story = {
  render: function Render() {
    const [value, setValue] = useState("");
    return (
      <TextInput
        label="Label"
        value={value}
        onChange={(event) => setValue(event.target.value)}
        invalid
      />
    );
  },
};
