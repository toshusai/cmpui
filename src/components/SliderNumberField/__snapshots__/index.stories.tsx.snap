// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/Form/SliderNumberField Basic smoke-test 1`] = `
<div class="cmpui_text-input__root cmpui_slider-number-input__root">
  <label class="cmpui_text-input__label cmpui_slider-number-input__label">
    value
  </label>
  <input type="text"
         class="cmpui_text-input__input"
         placeholder
         inputmode="decimal"
         value="0"
  >
</div>
`;

exports[`Components/Form/SliderNumberField Mixed smoke-test 1`] = `
<div>
  <div class="cmpui_text-input__root cmpui_slider-number-input__root">
    <label class="cmpui_text-input__label cmpui_slider-number-input__label">
      value
    </label>
    <input type="text"
           class="cmpui_text-input__input"
           placeholder
           value="mixed"
    >
  </div>
  <div style="display: flex; flex-direction: column; gap: 4px; margin-top: 8px;">
    <div class="cmpui_text-input__root">
      <input type="number"
             class="cmpui_text-input__input"
             placeholder
             value="1"
      >
    </div>
    <div class="cmpui_text-input__root">
      <input type="number"
             class="cmpui_text-input__input"
             placeholder
             value="2"
      >
    </div>
    <div class="cmpui_text-input__root">
      <input type="number"
             class="cmpui_text-input__input"
             placeholder
             value="3"
      >
    </div>
  </div>
</div>
`;
